tools/rosbag/src/recorder.cpp

```diff
@@ -285,9 +302,25 @@ std::string Recorder::timeToStr(T ros_t)
 void Recorder::doQueue(const ros::MessageEvent<topic_tools::ShapeShifter const>& msg_event, string const& topic, shared_ptr<ros::Subscriber> subscriber, shared_ptr<int> count) {
     //void Recorder::doQueue(topic_tools::ShapeShifter::ConstPtr msg, string const& topic, shared_ptr<ros::Subscriber> subscriber, shared_ptr<int> count) {
     Time rectime = Time::now();
+    std::string topic_name = subscriber->getTopic();
+
+    if(options_.custom_freq && custom_record_freq_.find(topic_name) != custom_record_freq_.end()){
+        ros::Duration interval = ros::Duration(double(1)/custom_record_freq_.at(topic_name));
+        if(topic_time_catcher_.find(topic_name) == topic_time_catcher_.end()){
+            topic_time_catcher_.emplace(topic_name, rectime + interval);
+        }
+        else if(rectime > topic_time_catcher_.at(topic_name)){
+            // correctoness in interval
+            interval -= (rectime - topic_time_catcher_.at(topic_name));
+            topic_time_catcher_.insert_or_assign(topic_name, rectime + interval);
```

Comment:
could just do `topic_time_catcher_.at(topic_name) + interval`

Reply:
Could drop insert_or_assign and just increment the value of the time_catcher by interval. You can drop c++17 dependency with this